app/assets/javascripts/modules/logger.js

```diff
@@ -0,0 +1,30 @@
+(function () {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  function Logger() {
+    if (isDevelopmentEnvironment()) {
+      return console;
+    } else {
+      return silentConsole();
+    }
+  }
+
+  function isDevelopmentEnvironment() {
+    return /\.dev\.gov\.uk$/.test(window.location.hostname);
+  }
+
+  function silentConsole() {
+    var fakeConsole = {};
+
+    $.each(console, function(key) {
+      if (typeof console[key] === 'function') {
+        fakeConsole[key] = function() {};
+      }
+    });
+
+    return fakeConsole;
+  }
+
+  GOVUK.Logger = Logger;
+})();
```

Comment:
Definitely the latter, but I'm not too fussed I guess; the code is now written this way to be far more explicit about:

- which methods are _actually_ used by external JavaScript (we previously exposed **everything** and figuring out what was actually being used externally was pretty hard to see)
- make it more explicit what we're trying to track when setting custom dimensions by passing them directly into the tracking event/pageview they're intended for

In terms of how people actually want to use this, though, we could just leave it open?